dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/data/NeuralNetworkDataParser.java

```diff
@@ -83,13 +90,33 @@ public void parse() {
         final Matrix data = input.get(range(0, input.getColumns() - 2));
         final int label = (int) input.get(input.getColumns() - 2);
         final boolean isValidation = ((int) input.get(input.getColumns() - 1) == 1);
-        trainingData.add(new Pair<>(new Pair<>(data, label), isValidation));
+
+        if (isValidation) {
+          final Pair<Matrix, int[]> batch = validationBatchGenerator.push(data, label);
```

Comment:
As you mentioned, it will be some benefits not to have separate class for generating batches. However, without `BatchGenerator`, a code for data and label lists for training and validation inputs, and handling these lists, will look too complicated. That's why I introduced `BatchGenerator` to make code more clear.
Anyway, I will modify `push` method that returns a value and make `BatchGenerator` not static in order to use `matrixFactory` and `batchSize` directly.